src/cli/telemetry.py

```
@@ -5,13 +5,58 @@
 from _compat.structlog import get_logger
-from prometheus_client import CollectorRegistry, Counter, Histogram, generate_latest
+
+_PROMETHEUS_AVAILABLE = importlib.util.find_spec("prometheus_client") is not None
+
+if _PROMETHEUS_AVAILABLE:  # pragma: no branch - import-time check
+    from prometheus_client import CollectorRegistry, Counter, Histogram, generate_latest
+else:
+    class CollectorRegistry:  # type: ignore[no-redef]
+        """No-op CollectorRegistry replacement when prometheus_client is unavailable."""
+
+        def __init__(self, *args: Any, **kwargs: Any) -> None:  # noqa: D401 - simple shim
```

Ruff (0.13.2)

Unused `noqa` directive (non-enabled: `D401`)

Remove unused `noqa` directive

(RUF100)